go/teams/chain.go

```diff
@@ -2214,6 +2237,50 @@ func (t *teamSigchainPlayer) obsoleteInvites(stateToUpdate *TeamSigChainState, r
 	}
 }
 
+func (t *teamSigchainPlayer) useInvites(stateToUpdate *TeamSigChainState, roleUpdates chainRoleUpdates, used []SCMapInviteIDUVPair) error {
+	for _, pair := range used {
+		inviteID, err := pair.InviteID.TeamInviteID()
+		if err != nil {
+			return err
+		}
+		invite, ok := stateToUpdate.inner.ActiveInvites[inviteID]
+		if !ok {
+			return fmt.Errorf("could not find active invite ID in used_invites: %s", inviteID)
+		}
+		if invite.MaxUses == nil {
+			return fmt.Errorf("`used_invites` for an invite that did not have `max_uses`: %s", inviteID)
+		}
+		uses := len(stateToUpdate.inner.UsedInvites[inviteID])
+		if uses+1 >= *invite.MaxUses {
+			return fmt.Errorf("invite %s is expired after %d uses", inviteID, uses+1)
+		}
+		uv, err := keybase1.ParseUserVersion(pair.UV)
+		if err != nil {
+			return err
+		}
+		var foundUV bool
+		for _, updatedUV := range roleUpdates[invite.Role] {
+			if uv.Eq(updatedUV) {
+				foundUV = true
+				break
+			}
+		}
+		if !foundUV {
+			return fmt.Errorf("used_invite for UV %s that was not added as role %s", pair.UV, invite.Role.HumanString())
+		}
+		logPoint := len(stateToUpdate.inner.UserLog[uv]) - 1
+		if logPoint < 0 {
+			return fmt.Errorf("used_invite for UV %s that was not added to to the team", pair.UV)
+		}
+		stateToUpdate.inner.UsedInvites[inviteID] = append(stateToUpdate.inner.UsedInvites[inviteID],
+			keybase1.TeamUsedInviteLog{
+				Uv:       uv,
+				LogPoint: logPoint,
+			})
```

Comment:
Should you also delete from ActiveInvites when the counter hits `max_uses`?

Reply:
I wasn't planning to but makes sense.

Reply:
Probably want them to disappear from lists in the UI when used up. Moving them out of ActiveInvites is likely an easy and long-term efficient way to do that

Reply:
However, if you throw it out then usedInvites[][].logPoint would be a bit less useful since the metadata of, for example, who created the invite would be gone. You could move them to `obsoleteInvites`. Or just trash them, the same as is done for existing invites.

Reply:
oh right, there is also a list of expired / used invite links in the GUI though.

Reply:
What do we do with time based invites? Every time the team is loaded we check if any expired by time and move them over (using time from the latest merkle root to prevent being misled by a bad os clock)? Or do we check which are expired every time we load the screen?

Reply:
I think we do nothing in the player for now. We can figure this out later.

I think for now:
- check expiration times on the server in helpers/team.iced when processing a change_membership link
- check expiration times in the client, also when handling resolutions
   - this changes semantics on few things (one that i know of, so maybe there's only one): when you add someone normally, through AddMember, it tries to complete outstanding social invites for that person as well. It should just skip expired ones, so it doesn't generate sketchy sigs (not "invalid sigs" because they are still valid from player's perspective, but would be rejected by server).
- check expiration time when in join request handler.

also I think for now, to keep things simple, we should disallow (on the server) `etime` for anything other than seitan_invite_token.